cassandra-bigtable-migration-tools/cassandra-bigtable-proxy/bigtable/bigtable.go

```
@@ -468,10 +504,58 @@ func (btc *BigtableClient) AlterTable(ctx context.Context, data *translator.Alte
 	return nil
 }
 
+// maybeAddCounterColumnFamily - adds the counter column family if: any or the provided columns are of the counter type and the schema mapping table doesn't show
+func (btc *BigtableClient) maybeAddCounterColumnFamily(ctx context.Context, adminClient *bigtable.AdminClient, keyspace string, tableName string, addColumns []message.ColumnMetadata) error {
+	addsCounterColumn := slices.ContainsFunc(addColumns, func(c message.ColumnMetadata) bool {
+		return c.Type == datatype.Counter
+	})
+	// apparently no counter column is being added, so we don't need to add the column family
+	if !addsCounterColumn {
+		return nil
+	}
+
+	alreadyHasCounterColumn := slices.ContainsFunc(btc.SchemaMappingConfig.GetAllColumns(keyspace, tableName), func(c *types.Column) bool {
+		return c.CQLType == datatype.Counter
+	})
+	// the table already has a counter column, so it's safe to assume the column family exists
+	if !alreadyHasCounterColumn {
```

Should this **not** be negated? i.e. if it already exists do nothing.

Also if the counter column family already exists but is of a different name than we are trying to add should we return an error?